### ISSUE TITLE
Reading/writing of YAML files is broken

### ISSUE THREAD

**Issue by mirkobunse** [JuliaData/YAML.jl#108]
Consider a dead-simple YAML file with a dollar sign in a String:

```
shell> cat input.yml
property: "Simple $ dollar sign"

julia> c = YAML.load_file("input.yml")
Dict{Any,Any} with 1 entry:
  "property" => "Simple \$ dollar sign"
```

Looks like reading the file has added an escape character. Printing the String, however, works as desired:

```
julia> println(c["property"])
Simple $ dollar sign
```

Now let's write back to YAML:

```
julia> YAML.write_file("output.yml", c)

shell> cat output.yml
property: "Simple \$ dollar sign"
```

It prints the escape character, which breaks the reading. You can not read with YAML.jl what YAML.jl has emitted:

```
julia> YAML.load_file("output.yml")
ERROR: while scanning a double-quoted scalar at line 1, column 10: found unknown escape character '$' at line 1, column 18
```

**Comment by mirkobunse** [comment]
The error does not appear with YAML.jl `v0.4.2`, but with all subsequent versions.

**Comment by kescobo** [comment]
Seems like #89 or #90 must be the culprit - those are the primary PRs merged between 4.2 and 4.3

**Comment by kescobo** [comment]
Oh, I guess just #90, since 89 was closed without merging
